### PR TITLE
remove derivatives from well rate computations

### PR DESCRIPTION
**Pull request by totto82**
This adds Scalar version of computePerfRate.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
Maybe I noticed the difficulties before. Hopefully the code duplication can be minimized through template or other approach.

**Comment by totto82** [comment]
> Maybe I noticed the difficulties before. Hopefully the code duplication can be minimized through template or other approach.

Agree. That's why I put it as draft. I would like to test it a bit before I start messing with the templates. I may need to move it back to the _impl file.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
> Hopefully the code duplication can be minimized through template or other approach.

I have minimized the code duplication. This should be ready for review and merge....

File: opm/simulators/wells/StandardWell_impl.hpp [hunk 1]
```diff
@@ -540,7 +769,78 @@ namespace Opm
...
+            if (!FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
+                OPM_DEFLOG_THROW(std::runtime_error, "Water is required when polymer is active", deferred_logger);
+            }
+
+            // for the cases related to polymer molecular weight, we assume fully mixing
+            // as a result, the polymer and water share the same viscosity
+            if constexpr (!Base::has_polymermw) {
+                std::vector<EvalWell> mob_eval(num_components_, {this->numWellEq_ + numEq, 0.});
+                updateWaterMobilityWithPolymer(ebosSimulator, perf, mob_eval, deferred_logger);
+                for (int i = 0; i < mob.size(); ++i) {
```
Comment (akva2) [inline opm/simulators/wells/StandardWell_impl.hpp:838]: size_t

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
@GitPaean @akva2 Any more comments? I hope to merge this soon.

**Comment by GitPaean** [comment]
I am not in the office now so having very limited capacity, so the comment is very brief.

I think the your way in handling `cmix_s` is pretty smart in term of avoiding making a few functions template. 

Ideally,  the function `computePerfRateEval` and `computePerfRateScalar` can be made to be one, basically, the code before the calling function `computePerfRate` needs to be one using template. Maybe @akva2 and @bska can give some suggestions about the best way to achieve this.

**Comment by totto82** [comment]
> Ideally, the function computePerfRateEval and computePerfRateScalar can be made to be one, basically, the code before the calling function computePerfRate needs to be one using template. Maybe @akva2 and @bska can give some suggestions about the best way to achieve this

I suggest we merge this now. Further re-factorization can be done later. This is a real performance hitter for cases with gaslift.

**Pull request merged**